client/src/lib/api.js

```diff
@@ -0,0 +1,231 @@
+ */
+export async function fetchGroupChartLogs(id) {
+  const response = await fetch(`/api/chart-group-exercises/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches user's logs from the exercises table in the API.
+ * @param {Number} 'id' - an integer representing the userId.
+ * @returns Promise that resolves to an array of exercise logs.
+ */
+export async function fetchUserLogs(id) {
+  const response = await fetch(`/api/exercises/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches user's groups from the groups table in the API.
+ * @param {Number} 'id' - an integer representing the userId.
+ * @returns Promise that resolves to an array of groups.
+ */
+export async function fetchGroups(id) {
+  const response = await fetch(`/api/groups/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches group exercises from the API.
+ * @param {Number} 'id' - an integer representing the groupId.
+ * @returns Promise that resolves to an array of exercise logs for all group members.
+ */
+export async function fetchGroupLogs(id) {
+  const response = await fetch(`/api/group-logs/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches group data from the API.
+ * @param {Number} 'id' - an integer representing the groupId.
+ * @returns Promise that resolves to an array of group settings data.
+ */
+export async function fetchGroupSettings(id) {
+  const response = await fetch(`/api/group-settings/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches group data from the API.
+ * @param {Number} 'id' - an integer representing the groupId.
+ * @returns Promise that resolves to an array of group penalties data.
+ */
+export async function fetchGroupPenalties(id) {
+  const response = await fetch(`/api/group-penalties/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches penalties data from the API.
+ * @param {Number} 'id' - an integer representing the userId.
+ * @returns Promise that resolves to an array of the user's penalties.
+ */
+export async function fetchUserPenalties(id) {
+  const response = await fetch(`/api/user-penalties/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches all login information and verifies account from the API.
+ * @returns Promise that resolves to an object with a token and user information.
+ */
+export async function postAccount(account) {
```

Comment:
I would probably name this `loginAccount` since that's the semantic operation it's doing.